File: cpp/prob_922.cpp
class Solution {
public:
    vector<int> sortArrayByParityII(vector<int>& A) {
        vector<int> even_list;
        vector<int> odd_list;
        vector<int> out_list;
        for (int i=0; i<A.size(); i++) {
            if (A[i] % 2 == 0) {
                even_list.push_back(A[i]);
            }
            else {
                odd_list.push_back(A[i]);
            };
        };
        for (int i=0; i<even_list.size(); i++) {
            out_list.insert(out_list.end(), {even_list[i], odd_list[i]});
        };
        return out_list;
    };
};

File: js/prob_922.js
/**
 * @param {number[]} A
 * @return {number[]}
 */
const sortArrayByParityII = A => {
    let even_list = []
    let odd_list = []
    let out_list = []
    
    for (let i=0; i<A.length; i++) {
        if (A[i] % 2 == 0) {
            even_list.push(A[i])
        }
        else {
            odd_list.push(A[i])
        }
    }
    for (let i=0; i<even_list.length; i++) {
        out_list.push(even_list[i], odd_list[i])
    }
    return out_list
}